File: TrabajoFinalPrueba1/PegasusV1/PegasusV1/Entities/CursoMateria.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace PegasusV1.Entities
{
    [Table("CURSO_MATERIA")]
    public class CursoMateria
    {
        public int Id { get; set; }

        [ForeignKey("Id_Materia")]
        public Materia? Materia { get; set; }
        public int? Id_Materia { get; set; }

        [ForeignKey("Id_Curso")]
        public Curso? Curso { get; set; }
        public int? Id_Curso { get; set; }
    }
}

File: pegasusapp/src/models/cursoMateria.ts
import { Materia } from './materia';
import { Curso } from './curso';

export interface CursoMateria {
  id: number;
  id_Materia?: number | null;
  id_Curso?: number | null;

  materia?: Materia | null;
  curso?: Curso | null;
}
